pkg/manager/member/scaler.go

```diff
@@ -230,3 +257,56 @@ func normalizeDeleteSlots(replicas int32, deleteSlots sets.Int32, desiredDeleteS
 	}
 	return deleteSlots
 }
+
+// setReplicasAndDeleteSlotsByFinished sets the replica and deleteSlots according to the ordinals finished successfully.
+// parameter means:
+// - scaling:
+//   - 1: scaling out
+//   - -1: scaling in
+// - ordinals: pod oridnals to create or delete this round, in processing order, which means increasing when scale out and decreasing when scale in.
+// - finishedOrdinals: oridnals finished successfully in this round.
+func setReplicasAndDeleteSlotsByFinished(scaling int, newSet, oldSet *apps.StatefulSet, ordinals []int32, finishedOrdinals sets.Int32) {
```

Comment:
How about adding some UTs for this function?

Reply:
Done. PTAL.